dcicutils/qa_utils.py

```diff
@@ -678,15 +682,403 @@ def Session(self, **kwargs):  # noQA - This name was chosen by AWS, so please do
         return _SessionModule(boto3=self)
 
 
+@MockBoto3.register_client(kind='session')
 class MockBoto3Session:
 
-    def __init__(self, *, region_name, boto3):
-        self.region_name = region_name
-        self.boto3 = boto3
+    _SHARED_DATA_MARKER = "_SESSION_SHARED_DATA_MARKER"
+
+    def __init__(self, *, region_name=None, boto3=None, **kwargs):
+        self.boto3 = boto3 or MockBoto3()
+
+        # These kwargs key names are the same as those for the boto3.Session() constructor.
+        self.aws_access_key_id = kwargs.get("aws_access_key_id")
+        self.aws_secret_access_key = kwargs.get("aws_secret_access_key")
+        self.aws_region = region_name
+
+        # These is specific for testing.
+        self.aws_credentials_dir = None
+
+    # FYI: Some things to note about how boto3 (and probably any AWS client) reads AWS credentials/region.
+    #  - It looks (of course) at envrionment variables before files.
+    #  - It wants access key ID and secret access key BOTH to come from the same source,
+    #    e.g. does not get access key ID from environment variable and secret access key from file.
+    #  - It reads region from EITHER the credentials file OR the config file, the former FIRST;
+    #    though (of course) it does NOT read access key ID or secret access key from the config file. 
+    #  - The aws_access_key_id, aws_secret_access_key, and region properties in the credentials/config
+    #    files may be EITHER upper AND/OR lower case; but the environment variables MUST be all upper case.
+    #  - If file environment variables (i.e. AWS_SHARED_CREDENTIALS_FILE, AWS_CONFIG_FILE) are NOT set,
+    #    i.e. SET to None, it WILL look at the default credentials/config files (e.g. ~/.aws/credentials);
+    #    which is why we set to /dev/null in unset_environ_credentials_for_testing().
+    #
+    # NOTE: The get_credentials method, region_name property, and related methods were
+    # added to support usage by 4dn-cloud-infra/setup-remaining-secrets unit tests (June 2022).
 
     def client(self, service_name, **kwargs):
         return self.boto3.client(service_name, **kwargs)
 
+    def unset_environ_credentials_for_testing(self) -> None:
+        """
+        Unsets any/all AWS credentials related environment variables for testing.
+        """
+        os.environ.pop("AWS_ACCESS_KEY_ID", None)
+        os.environ.pop("AWS_CONFIG_FILE", "/dev/null")
+        os.environ.pop("AWS_DEFAULT_REGION", None)
+        os.environ.pop("AWS_REGION", None)
+        os.environ.pop("AWS_SECRET_ACCESS_KEY", None)
+        os.environ.pop("AWS_SESSION_TOKEN", None)
+        os.environ.pop("AWS_SHARED_CREDENTIALS_FILE", "/dev/null")
```

Comment:
I'm using pop just because get a key exception doing `del os.environ[absent_key]`; do I don't need the return value from pop, but it's the only way I saw to delete without exception (without doing an extra `in` check). And the only issue doing it like you suggest is that the `_FILE` ones should be set to `/dev/null` not None, but I could check for that in the loop, or pass dictionary with values (none or `/dev/null`) as needed.